### PR TITLE
Report issues with metadata

### PR DESCRIPTION
**Pull request by mitchdawson1982**
This PR creates a service to raise issues with metadata on the FMD front end.

**Implements**
- new `/feedback/issue` url path
- new `report_issue_view` 
- new `ReportIssue` model for storing successfully submitted issues
- new `ReportIssueForm`  model form & html template for accepting and validating user input before generating a `ReportIssue` model instance
- new service module and `send_notifcations` function for interacting with the notify API
- new variables and secrets for the notify api key(s) and notification template id's
- new tests for forms, models and service

### Key Decisions
I have opted to provide the report issue form in a separate view and page via a button on the entity details page. 
Some examples I have seen that implement forms on an existing page do so at the bottom, which means the user needs to scroll to the bottom. 

I tested placing the form at the top of the entity details page, below the contact section, but it looked cluttered on top of the data. Implementing this way also avoids the need for js to show/hide and validate the form in the front end and allows us to utilise the existing backend methods of form & model validation on submission.

When the report issue form is requested via GET request, we obtain the `entity name, full URL and data owner` values from the request and store these on the user's session to be used to populate the issue model instance alongside the form values on submission. This avoids the need for creating additional template and form fields to hold these values and placing them back in front of the user. Successfully submitted issues are stored as model instances so that these can be reported on in the future if necessary.

I have implemented a basic integration with the GDS Notify service utilising their Python SDK in which we "fire and forget" notification jobs into the service, logging responses and exceptions as necessary. We could implement a more detailed implementation which periodically polls for message status updates, however, this would then require asynchronous jobs using something such as [Celery](https://docs.celeryq.dev/en/stable/getting-started/introduction.html) or [Python RQ](https://python-rq.org/), and would probably go further than our need at this point but is something we could look at in future.

### For further consideration

- Details page button location / style
image

- Location/ Styling of the entity name on the issue form
image

### CODE REVIEW

**Review by MatMoore — DISMISSED** [review]
Great work, and thanks for the detailed write up.

The only thing that jumps out at me now is the use of data owner, which we know is the wrong role to notify. But if we're OK revisiting that next sprint then we can ignore that for now 👍🏻

File: feedback/templates/report_issue.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+{% block content %}
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <h1 class="govuk-heading-xl">{{h1_value}}</h1>
+  </div>
+
+  <div class="govuk-grid-column-full">
+    <p class="govuk-body">
+      <a href="{{entity_url}}" class="govuk-link govuk-link--no-visited-state govuk-heading-l">{{entity_name}}</a>
```
[the remark this answers is not in the document]
Comment (mitchdawson1982) [reply feedback/templates/report_issue.html:13]: Yes I agree. I was thinking the same and should have added this to points for consideration but it slipped my mind.

File: feedback/views.py [hunk 2]
```diff
@@ -36,3 +39,53 @@ def thank_you_view(request) -> HttpResponse:
...
+    if request.method == "POST":
+        form = ReportIssueForm(request.POST)
+        if form.is_valid():
+            issue = form.save(commit=False)
+            issue.entity_name = request.session.get("entity_name")
+            issue.entity_url = request.session.get("entity_url")
+            issue.data_owner_email = request.session.get("data_owner_email")
+            issue.save()
+
+            if settings.NOTIFY_ENABLED:
```
[the remark this answers is not in the document]
Comment (mitchdawson1982) [reply feedback/views.py:54]: Makes sense.

File: feedback/models.py [hunk 3]
```diff
@@ -21,3 +24,37 @@ class Feedback(models.Model):
     how_can_we_improve = models.TextField(
         verbose_name=_("How can we improve this service?"), null=False, blank=True
     )
+
+
+class ReportIssue(models.Model):
+    class IssueChoices(models.TextChoices):
+        BROKEN_LINK = "Broken link"
+        INCORRECT_OWNER = "Incorrect owner"
+        OUTDATED_CONTACT = "Outdated contact"
```
[the remark this answers is not in the document]
Comment (mitchdawson1982) [reply feedback/models.py:33]: Agreed.

File: feedback/models.py [hunk 4]
```diff
@@ -21,3 +24,37 @@ class Feedback(models.Model):
...
+    id = models.UUIDField(
+        default=uuid.uuid4, unique=True, primary_key=True, editable=False
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    reason = models.CharField(
+        max_length=50,
+        choices=IssueChoices.choices,
+        verbose_name=_("Reason"),
```
[the remark this answers is not in the document]
Comment (mitchdawson1982) [reply feedback/models.py:45]: Makes sense.

File: feedback/models.py [hunk 5]
```diff
@@ -21,3 +24,37 @@ class Feedback(models.Model):
     how_can_we_improve = models.TextField(
         verbose_name=_("How can we improve this service?"), null=False, blank=True
     )
+
+
+class ReportIssue(models.Model):
```
[the remark this answers is not in the document]
Comment (mitchdawson1982) [reply feedback/models.py:29]: Makes sense.

File: feedback/views.py [hunk 6]
```diff
@@ -36,3 +38,52 @@ def thank_you_view(request) -> HttpResponse:
...
+            return redirect("feedback:thanks")
+
+        else:
+            log.error(f"Unexpected invalid report issue form submission: {form.errors}")
+            return render(
+                request,
+                "report_issue.html",
+                {
+                    "h1_value": _(
+                        f"Report an issue with {request.session.get('entity_name')}"
```
[the remark this answers is not in the document]
Comment (mitchdawson1982) [reply feedback/views.py]: Changed as per the example.

**Review by MatMoore — APPROVED** [review]

**Pull request merged**